test/parallel/test-worker-with-nested-async-stack-gets-terminated.js

```diff
@@ -0,0 +1,91 @@
+// Flags: --expose-internals
+
+'use strict';
+const common = require('../common');
+const { Worker } = require('worker_threads');
+
+/**
+ * This file is a minimal possible reproduction and an
+ * explanation of https://github.com/nodejs/node/issues/43084.
+ *
+ * It can not trigger the bug consistently because
+ * that would require a rare thread execution interleaving
+ * condition.
+ *
+ * test-worker-http2-stream-terminate.js has a bigger chance
```

Comment:
Is there any reason we need this new test that has less chance to trigger the bug?